src/main/java/ch/njol/skript/effects/EffDetonate.java

```diff
@@ -0,0 +1,80 @@
+package ch.njol.skript.effects;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.type.Fire;
+import org.bukkit.block.data.type.TNT;
+import org.bukkit.entity.*;
+import org.bukkit.entity.minecart.ExplosiveMinecart;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+
+@Name("Detonate Entity")
+@Description("Immediately detonates an entity or block. Accepted entities are fireworks, TNT minecarts, wind charges, creepers, and tnt. Accepted blocks are tnt.")
+@Examples("detonate last launched firework")
+@Since("INSERT VERSION")
+public class EffDetonate extends Effect {
+
+	private static final boolean HAS_WINDCHARGE = Skript.classExists("org.bukkit.entity.WindCharge");
+
+	static {
+		Skript.registerEffect(EffDetonate.class, "detonate %entities/blocks%");
+	}
+
+	private Expression<?> thingsToDetonate;
+
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		this.thingsToDetonate = (Expression<?>) exprs[0];
+ 		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		for (Object detonateThis : thingsToDetonate.getArray(event)) {
+			if (detonateThis instanceof Block && ((Block) detonateThis).getType() == Material.TNT) {
+				Block block = (Block) detonateThis;
+				Location location = block.getLocation();
+				World world = block.getWorld();
+				block.setType(Material.AIR);
+				TNTPrimed tnt = world.spawn(location, TNTPrimed.class);
+				tnt.setFuseTicks(0);
+			}
```

Comment:
I'm not sure allowing the TNT block is a good thing. This feels like we're making some kind of ersatz behaviour, which we try to avoid because it won't necessarily respect what the server would normally do: i.e. anything that usually interacts with this process (the interaction with the TNT block, listening to the spawning of the entity, etc.) might not happen as the user expects.
Since this is such a simple thing (set the tnt block to air, spawn a tnt entity) it's probably something the user ought to be doing for themselves rather than needing special handling in a syntax.